src/gui/wifilistmanager.cpp

```diff
@@ -0,0 +1,88 @@
+#include "wifilistmanager.h"
+#include "ui_wifilistmanager.h"
+#include "networksettings.h"
+
+#include "configfile.h"
+
+#include <QPushButton>
+
+namespace OCC {
+
+WifiListManager::WifiListManager(QWidget *parent)
+    : QDialog(parent)
+    , _ui(new Ui::WifiListManager)
+{
+    _ui->setupUi(this);
+
+    this->setAttribute(Qt::WA_DeleteOnClose);
+
+    loadListMode();
+    loadSsidList();
+
+    connect(_ui->saveExitBtn, &QPushButton::clicked, this, &WifiListManager::saveSsidSettings);
+    connect(_ui->cancelExitBtn, &QPushButton::clicked, [this]() { this->close(); });
+    connect(_ui->addToListBtn, &QPushButton::clicked, [this]() { WifiListManager::addListItem(_ui->ssidEntryEdit->text()); });
+    connect(_ui->deleteFromListBtn, &QPushButton::clicked, [this]() { WifiListManager::deleteListItems(_ui->ssidList->selectedItems()); });
+    connect(_ui->ssidEntryEdit, &QLineEdit::editingFinished, [this]() { WifiListManager::addListItem(_ui->ssidEntryEdit->text()); _ui->ssidEntryEdit->setText(QString("")); });
+}
+
+WifiListManager::~WifiListManager()
+{
+    delete _ui;
+}
+
+void WifiListManager::loadListMode()
+{
+    OCC::ConfigFile cfg;
+    auto test = cfg.wifiListMode();
+    (cfg.wifiListMode() == "whitelist") ? _ui->whitelistCheckbox->setChecked(true) : _ui->blacklistCheckbox->setChecked(true);
+}
+
+void WifiListManager::loadSsidList()
+{
+    OCC::ConfigFile cfg;
+    QStringList ssidLoadedList = cfg.ssidList();
+    if ( !(ssidLoadedList.count() == 1 && ssidLoadedList.at(0).isEmpty()) ) {
```

Comment:
Please drop the top negation for readability, also please avoid the spaces within the parentheses. This would give:
`if (ssidLoadedList.count() != 1 || ssidLoadedList.at(0).isEmpty()) {`